lib/install.mjs

```diff
@@ -0,0 +1,132 @@
+  let prerelease = parsedVersion.prerelease[0].split('-');
+  if (prerelease.length > 1) {
+    prerelease = prerelease.slice(0, -1);
+  }
+  versionString = `${versionString}-${prerelease}`;
+}
+
+// Check build flavor
+if (versionString.slice(-4) === '-sdk') {
+  versionString = versionString.slice(0, -4);
+  buildType = 'sdk';
+} else if (versionString.slice(-3) === 'sdk') {
+  versionString = versionString.slice(0, -3);
+  buildType = 'sdk';
+}
+
+let url = '';
+let hostArch = env.npm_config_nwjs_process_arch || arch;
+let urlBase = env.npm_config_nwjs_urlbase || env.NWJS_URLBASE || 'https://dl.nwjs.io/v';
+let buildTypeSuffix = buildType === 'normal' ? '' : `-${buildType}`;
+let hostOs = env.npm_config_nwjs_platform || env.NWJS_PLATFORM || platform;
+
+// Determine download url
+switch (hostOs) {
+  case 'win32':
+    url = `${urlBase}${versionString}/nwjs${buildTypeSuffix}-v${versionString}-win-${hostArch}.zip`;
+    break;
+  case 'darwin':
+    url = `${urlBase}${versionString}/nwjs${buildTypeSuffix}-v${versionString}-osx-${hostArch}.zip`;
+    break;
+  case 'linux':
+    url = `${urlBase}${versionString}/nwjs${buildTypeSuffix}-v${versionString}-linux-${hostArch}.tar.gz`;
+    break;
+}
+
+if (url === '') {
+  console.error('[ERROR] Could not find a compatible version of nw.js to download for your platform.');
+}
+
+const __dirname = dirname(fileURLToPath(import.meta.url));
+let dest = resolve(__dirname, '..', 'nwjs');
+
+if (existsSync(dest) === true) {
+  console.info("[ INFO ] The NW.js binary already exists.");
+  exit(0);
+}
+
+let parsedUrl = new URL(url);
+let filePath = '';
+if (parsedUrl.protocol === 'file:') {
+  filePath = resolve(
+    decodeURIComponent(
+      url.slice('file://'.length)
+    )
+  );
+
+  if (existsSync(filePath) === false) {
+    console.error('[ERROR] Could not find ' + filePath);
+  }
+
+} else {
+
+  const bar = new progress.SingleBar({}, progress.Presets.rect);
+
+  const stream = createWriteStream(dest);
+
+  get(url, (response) => {
+
+    let chunks = 0;
+    bar.start(Number(response.headers['content-length']), 0);
+    response.on('data', async (chunk) => {
+      chunks += chunk.length;
+      bar.increment();
+      bar.update(chunks);
+    });
+
+    response.on('error', () => {
+      rmSync(dest, {
+        recursive: true,
+        force: true,
+      });
+    });
+
+    response.on('end', () => {
+      if (platform === 'linux') {
+        compressing.tgz.uncompress(dest, '.')
+          .then(() => rmSync(dest, {
+            recursive: true,
+            force: true,
+          }))
+          .then(() => {
+            renameSync(`nwjs-v${versionString}-${platform}-${arch}`, 'nwjs');
+          });
+      } else {
+        compressing.zip.uncompress(dest, '.')
+          .then(() => rmSync(dest, {
+            recursive: true,
+            force: true,
+          }))
+          .then(() => {
+            renameSync(`nwjs-v${versionString}-${platform}-${arch}`, 'nwjs');
+          });
```

Comment:
Abstracted out repeated calls to functions to make code DRY. Since both calls to `compressing` were identical, except for the compression type, I simplified that as well, so they can't get out-of-sync.
```suggestion
    function deleteDestinationFolder () {
      const deleteOptions = {
        recursive: true,
        force: true
      };
      rmSync(dest, deleteOptions);
    }

    response.on('error', async () => {
      await deleteDestinationFolder();
    });

    response.on('end', () => {
      let compressionType = 'zip';
      if (hostOs === 'linux') {
        compressionType = 'tgz';
      };
      compressing[compressionType].uncompress(dest, '.')
        .then(async () => {
          await deleteDestinationFolder();
        })
        .then(async () => {
          await renameSync(`nwjs-v${versionString}-${platform}-${arch}`, 'nwjs');
        })
        .catch((err) => {
          // Do some type of error handling.
        });
```

Reply:
Same comment here - used most of this code but not all of it. Still have some small changes to make though.